NitroxClient/MonoBehaviours/Gui/MainMenu/MainMenuMultiplayerPanel.cs

```diff
@@ -78,55 +84,48 @@ public void CreateServerButton(string text, string joinIp, string joinPort)
                 OpenJoinServerMenu(joinIp, joinPort);
             });
 
-            GameObject delete = Instantiate(SavedGamesRef.GetComponent<MainMenuLoadPanel>().saveInstance.GetComponent<MainMenuLoadButton>().deleteButton);
+            GameObject delete = Instantiate(deleteButtonRef, multiplayerButtonInst.transform, false);
             Button deleteButtonButton = delete.GetComponent<Button>();
             deleteButtonButton.onClick = new Button.ButtonClickedEvent();
             deleteButtonButton.onClick.AddListener(() =>
             {
                 RemoveServer(multiplayerButtonInst.transform.GetSiblingIndex() - 1);
                 Destroy(multiplayerButtonInst);
             });
-            delete.transform.SetParent(multiplayerButtonInst.transform, false);
         }
 
-        public void AddServer(string name, string ip, string port)
+        private void AddServer(string name, string ip, string port)
         {
             using (StreamWriter sw = new StreamWriter(SERVER_LIST_PATH, true))
             {
                 sw.WriteLine($"{name}|{ip}|{port}");
             }
         }
 
-        public void RemoveServer(int index)
+        private void RemoveServer(int index)
         {
             List<string> serverLines = new List<string>(File.ReadAllLines(SERVER_LIST_PATH));
             serverLines.RemoveAt(index);
             File.WriteAllLines(SERVER_LIST_PATH, serverLines.ToArray());
         }
 
-        public void OpenJoinServerMenu(string serverIp, string serverPort)
+        public static void OpenJoinServerMenu(string serverIp, string serverPort)
         {
+            if (main == null)
+            {
+                return;
```

Comment:
Should throw exception because this is an important function. If it fails, program doesn't work right.